20221018_homework04.py

```diff
@@ -0,0 +1,39 @@
+# Задача 0004
+# Задана натуральная степень k. Сформировать случайным образом список
+# коэффициентов (значения от 0 до 100) многочлена и записать в файл многочлен степени k.
+# Пример:
+# k=2 => 2x^2 + 4x + 5 = 0 или x^2 + 5 = 0 или 10x = 0
+
+import sys
+from random import randint
+from sympy import Symbol, Poly
+
+
+for j in [1, 2]:  # цикл для записи двух файлов
+
+    try:
+        k = int(input('Введите степень k: '))
+    except ValueError:
+        print('Допускаются только натуральные числа. Программа завершает работу... ')
+        sys.exit()
+    if k < 1:
+        print('Допускаются только положительные числа. Программа завершает работу... ')
+        sys.exit()
+
+    # чтобы получить многочлен k-степени, индекс с номером 0 не должен быть нулем:
+    array = [randint(1,100)] + [str(randint(0, 100)) for i in range(k)]
```

Comment:
```suggestion
    array = [str(randint(0, 100)) for i in range(k + 1)]
```
В условии ведь четко указано, что коэффициенты могут принимать значения от нуля до 100. Поэтому не нужно упрощать себе работу. Буду тестировать на всех возможных коэффициентах.